File: solutions/ruby/run_solution.rb
require 'colorize'
require './repeated_string'

#input parsing and reading
input00 = File.open("../../test-cases/input/input00.txt")
input01 = File.open("../../test-cases/input/input01.txt")
input02 = File.open("../../test-cases/input/input02.txt")

data = input00.map(&:chomp)

s00 = data[0]
n00 = data[1].to_i

data = input01.map(&:chomp)

s01 = data[0]
n01 = data[1].to_i

data = input02.map(&:chomp)

s02 = data[0]
n02 = data[1].to_i

# problem application
start = Time.now
response00 = repeated_string s00, n00
response01 = repeated_string s01, n01
response02 = repeated_string s02, n02
finish = Time.now

puts("solved all the test cases in " + (1000 * (finish - start)).round(2).to_s + "ms")

#expected outputs
start = Time.now
output00File = File.open("../../test-cases/output/output00.txt")
output01File = File.open("../../test-cases/output/output01.txt")
output02File = File.open("../../test-cases/output/output02.txt")

data = output00File.map(&:chomp)
output00 = data[0].to_i

data = output01File.map(&:chomp)
output01 = data[0].to_i

data = output02File.map(&:chomp)
output02 = data[0].to_i

if response00 === output00
	puts "INPUT 00".green
else
	puts "INPUT 00".red
end

if response01 === output01
	puts "INPUT 01".green
else
	puts "INPUT 01".red
end

if response02 === output02
	puts "INPUT 02".green
else
	puts "INPUT 02".red
end

File: solutions/java/RunSolution.java
import java.io.*; 
import java.util.Arrays;
import java.math.BigInteger;

public class RunSolution{

	public static final String ANSI_RED = "\u001B[31m";
	public static final String ANSI_GREEN = "\u001B[32m";
	public static final String ANSI_RESET = "\u001B[0m";

	public static void main(String[] args) throws Exception{

		//input parsing and reading
		File input00File = new File("../../test-cases/input/input00.txt");
		BufferedReader brInput00 = new BufferedReader(new FileReader(input00File));

		File input01File = new File("../../test-cases/input/input01.txt");
		BufferedReader brInput01 = new BufferedReader(new FileReader(input01File));

		File input02File = new File("../../test-cases/input/input02.txt");
		BufferedReader brInput02 = new BufferedReader(new FileReader(input02File));

		String st;
		int currentLine = 0;
		String lineSplit[];

		String s00 = "";
		BigInteger n00 = null;

		String s01 = "";
		BigInteger n01 = null;

		String s02 = "";
		BigInteger n02 = null;

		while((st = brInput00.readLine()) != null){
			if(currentLine == 0){
				s00 = st;
			}else if(currentLine == 1){
				n00 = new BigInteger(st);
			}
			currentLine++;
		}

		currentLine = 0;

		while((st = brInput01.readLine()) != null){
			if(currentLine == 0){
				s01 = st;
			}else if(currentLine == 1){
				n01 = new BigInteger(st);
			}
			currentLine++;
		}

		currentLine = 0;

		while((st = brInput02.readLine()) != null){
			if(currentLine == 0){
				s02 = st;
			}else if(currentLine == 1){
				n02 = new BigInteger(st);
			}
			currentLine++;
		}

		// problem application

		BigInteger response00 = RepeatedString.run(s00, n00);
		BigInteger response01 = RepeatedString.run(s01, n01);
		BigInteger response02 = RepeatedString.run(s02, n02);
		

		// expected outputs

		File output00File = new File("../../test-cases/output/output00.txt");
		BufferedReader brOutput00 = new BufferedReader(new FileReader(output00File));

		File output01File = new File("../../test-cases/output/output01.txt");
		BufferedReader brOutput01 = new BufferedReader(new FileReader(output01File));

		File output02File = new File("../../test-cases/output/output02.txt");
		BufferedReader brOutput02 = new BufferedReader(new FileReader(output02File));

		BigInteger output00 = null;
		BigInteger output01 = null;
		BigInteger output02 = null;

		while((st = brOutput00.readLine()) != null){
			output00 = new BigInteger(st.trim());
		}

		while((st = brOutput01.readLine()) != null){
			output01 = new BigInteger(st.trim());
		}

		while((st = brOutput02.readLine()) != null){
			output02 = new BigInteger(st.trim());
		}


		System.out.println(((response00.equals(output00)) ? ANSI_GREEN : ANSI_RED) + "INPUT 00");
		System.out.println(((response01.equals(output01)) ? ANSI_GREEN : ANSI_RED) + "INPUT 01");
		System.out.println(((response02.equals(output02)) ? ANSI_GREEN : ANSI_RED) + "INPUT 02" + ANSI_RESET);
	}
}